build/scripts/build-utils.ps1

```diff
@@ -0,0 +1,41 @@
+# Collection of powershell build utility functions that we use across our scripts.
+
+Set-StrictMode -version 2.0
+$ErrorActionPreference="Stop"
+
+# Declare a number of useful variables for other scripts to use
+[string]$repoDir = Resolve-Path (Join-Path $PSScriptRoot "..\..")
+[string]$binariesDir = Join-Path $repoDir "Binaries"
+[string]$scriptDir = $PSScriptRoot
+
+function Create-Directory([string]$dir) {
+    New-Item $dir -ItemType Directory -ErrorAction SilentlyContinue | out-null
```

Comment:
Then presumably the next command which uses the directory will fail 😄 

I don't meant to be trolly there but there isn't a good answer to this question.  Asking if the directory actually failed to create or already exists is just a variation of the [File.Exists problem](https://blogs.msdn.microsoft.com/jaredpar/2009/12/10/the-file-system-is-unpredictable/).  I could test for existence of the directory after the call but it's a racy check.